log-base/src/Log/Class.hs

```diff
@@ -39,6 +39,8 @@ class MonadTime m => MonadLog m where
   localData   :: [Pair] -> m a -> m a
   -- | Extend the current application domain locally.
   localDomain :: T.Text -> m a -> m a
+  -- | Capture current context of a logger for use in the 'IO' monad.
+  getLoggerIO :: m (UTCTime -> LogLevel -> T.Text -> Value -> IO ())
```

Comment:
This makes `MonadLog` unimplementable for Non-IO based monads.

I'd rather prefer separate class, e.g. `MonadLogIO` (though `NoLogT m` could still have an instance for all `m`). That wouldn't require major bump, and be a little more flexible.

Reply:
Hmm, I see. I'd like to avoid additional class. How about replacing `getLoggerIO` with `getLoggerEnv :: m LoggerEnv`? It's more general as you can use it in conjunction with `logMessageIO` to get IO logging, but doesn't introduce any IO into the class.

Reply:
@arybczak that sounds :+1: if it works for you, though it will still be breaking change :)